packages/cursorless-engine/src/languages/TreeSitterQuery/rewriteStartOfEndOf.test.ts

```diff
@@ -0,0 +1,61 @@
+import { Range } from "@cursorless/common";
+import { MutableQueryCapture } from "./QueryCapture";
+import { SyntaxNode } from "web-tree-sitter";
+import { rewriteStartOfEndOf } from "./rewriteStartOfEndOf";
+import assert = require("assert");
+
+type NameRange = Omit<MutableQueryCapture, "allowMultiple" | "node">;
+
+interface TestCase {
+  name: string;
+  captures: NameRange[];
+  want: NameRange[];
+}
+
+const testCases: TestCase[] = [
+  {
+    name: "should rewrite startof to start of range",
+    captures: [
+      { name: "@value.iteration.start.startof", range: new Range(1, 2, 1, 3) },
+    ],
+    want: [{ name: "@value.iteration.start", range: new Range(1, 2, 1, 2) }],
+  },
+
+  {
+    name: "should rewrite endof to start of range",
+    captures: [
+      { name: "@value.iteration.start.endof", range: new Range(1, 2, 1, 3) },
+    ],
+    want: [{ name: "@value.iteration.start", range: new Range(1, 3, 1, 3) }],
+  },
+
+  {
+    name: "should leave other captures alone",
+    captures: [
+      { name: "@value.iteration.start", range: new Range(1, 2, 1, 3) },
+    ],
+    want: [{ name: "@value.iteration.start", range: new Range(1, 2, 1, 3) }],
+  },
+];
```

Comment:
i'd add tests for `@value.startOf` (without `.start`)

Reply:
done